libcloud/compute/drivers/dimensiondata.py

```diff
@@ -638,12 +639,18 @@ def ex_power_off(self, node):
         """
         request_elm = ET.Element('powerOffServer',
                                  {'xmlns': TYPES_URN, 'id': node.id})
-        body = self.connection.request_with_orgId_api_2(
-            'server/powerOffServer',
-            method='POST',
-            data=ET.tostring(request_elm)).object
-        response_code = findtext(body, 'responseCode', TYPES_URN)
-        return response_code in ['IN_PROGRESS', 'OK']
+
+        try:
+            body = self.connection.request_with_orgId_api_2(
+                'server/powerOffServer',
+                method='POST',
+                data=ET.tostring(request_elm)).object
+            response_code = findtext(body, 'responseCode', TYPES_URN)
+        except (DimensionDataAPIException, NameError, BaseHTTPError):
+            r = self.ex_get_node_by_id(node.id)
+            response_code = r.state.upper()
+
+        return response_code in ['IN_PROGRESS', 'OK', 'STOPPED', 'STOPPING']
```

Comment:
Thx @Kami  for your comment. My intension is to make it more user friendly; As long as the aim is achieved, doesn't matter if the state a precondition or the result of the operation, as long as it is 'STOPPED',  I'm happy, no exception. 
 
Throwing exception will means one will have to check the state of the every time. More code.